pmd-java/src/main/resources/category/java/errorprone.xml

```diff
@@ -702,29 +702,27 @@ public String bar(String string) {
             <property name="xpath">
                 <value>
 <![CDATA[
-//MethodDeclaration[
-  @Name='onCreate' or
-  @Name='onConfigurationChanged' or
-  @Name='onPostCreate' or
-  @Name='onPostResume' or
-  @Name='onRestart' or
-  @Name='onRestoreInstanceState' or
-  @Name='onResume' or
-  @Name='onStart'
-  ]
-    /Block[not(
-      (BlockStatement[1]/Statement/StatementExpression/PrimaryExpression[./PrimaryPrefix[@SuperModifier= true()]]/PrimarySuffix[@Image= ancestor::MethodDeclaration/@Name]))]
-[ancestor::ClassOrInterfaceDeclaration[ExtendsList/ClassOrInterfaceType[
-  pmd-java:typeIs('android.app.Activity') or
-  pmd-java:typeIs('android.app.Application') or
-  pmd-java:typeIs('android.app.Service')
-]]]
+//ClassOrInterfaceDeclaration
+    [
+      pmd-java:typeIs('android.app.Activity') or
```

Comment:
I think the description of the rule should mention that this is android-specific. I'll add that to the backlog of https://github.com/pmd/pmd/projects/3